csharp/src/Ice/Protocol.cs

```diff
@@ -78,4 +77,16 @@ public static Protocol Parse(string str)
             }
         }
     }
+
+    internal static class ProtocolDefinitions
+    {
+        internal static byte GetFrameType(this ReadOnlySpan<byte> header) => header[8];
```

Comment:
we will add a protocol parameter if needed later

Reply:
It's also unclear to me why the protocol trace needs to decode the frames' bytes. Why not trace the frame directly? This way, no need to read any bytes inside tracing. Do we need to trace before we create the frames?

Reply:
This frame type is only known if you read the headers, maybe we should change this, but currently we trace close connection, validate connection, request, batch request, response. The info is only available reading the headers.

Likewise we trace the compression status and request-id that are only in the headers, then we read the size from the header because the frame Size property is only the size of the payload but the trace size includes the headers size.